### ISSUE TITLE
Status of Aizoban Project

### ISSUE THREAD

**Issue by Adamlord1** [jparkie/Aizoban#48]
Guys please tell me any of u continuing its development .
And how can I remain connected to you all.
Thanks hope for reply.
I don't want this app to end please guys don't give up.

**Comment by Chogos** [comment]
Hi! I was planning to continue its development in early May or maybe even sooner, depending on my current project. If you're interested in contacting me here's my mail: tomislav.blazeka@gmail.com

**Comment by Adamlord1** [comment]
Okies brother 
Anyway so I can hope its development will start in April last right?
What's the status of app right now is everything working?
Well I am no developer but I can help with ideas,look designing and promotional tips I used to do it with jparkie.
Anyway why did it get removed from play store?

**Comment by jparkie** [comment]
A DMCA request was filed against the application with no appeal. It was suspended automatically by Google. I'm on a hiatus for development right now, but Chogos if you want, I can upload my current progress on Version 2.0.+. It has mostly everything done except for the MangaActivity and ChapterActivity. The PageFragment is done though. I made it more modular, more mockable, more futureproof, and the like. The only issue is backward compatibility with Version 1.0 databases.

**Comment by jparkie** [comment]
Accordingly,the query manager is pure CRUD-based and utilizes efficient ReentrantReadWriteLocks on `Update()`. Thus, multiple threads can `Create()` and `Read()` now while `Update()` and `Delete()` blocks.

I abandoned Cursors in favour of buffered POJO Streams to leverage the efficiency of RecyclerView. Initially for N items, it has a maximum of 2N space complexity, however, when the stream completes, it competes at N space complexity. As a result, it has similar final space complexity as a Cursor, better time complexity, and no overhead of maintaining an underlying native CursorWindow.

**Comment by jparkie** [comment]
The PageFragment utilizes a library now instead of my custom ImageView to be able to render extremely large images like Google Maps.

**Comment by Chogos** [comment]
Yes, I'd be very interested in your current progress. It's really a shame that the app got taken down.

**Comment by jparkie** [comment]
When I have time sometime this week, I'll polish some parts and upload it to a new branch. There are a lot of other improvements compared to the Version 1.0 series. The only issue if you continue the application is that compatibility with Version 1.0 databases really prevents the best it could be.

I made bad decisions for Version 1.0 as my first Android application, so if you want you can change it a lot and even brand it anew for you.

**Comment by Chogos** [comment]
Ok, there's no rush as I'm pretty busy myself currently.

Seeing that there are other, similar apps on the Play store which are even more popular I see no reason why Aizoban couldn't return one day. But we can worry about that later.

**Comment by Adamlord1** [comment]
Guys so we can slowly and steadily bring aizoban back OK.😂
Brother jparkie will u be back,how long u will stay on hiatus?
Anyway guys let's set an imaginary date where u all can get back to work.hehehe
I hope we can bring it back
My assumumption of removal are:
1)simple its work of mangarock dudes,why simple why do u think every manga reader got removed just manga rock stayed. It is a basic strategy many big companies do it too ,u might have heard abt Samsung vs apple battles. That is a high possibility.
2)licensed series but this reason is unlikely.

Can u tell me what is this automated app blocker google has.
Did u appeal to google to bring back app.
I think u can win easily if DMCA appeal was filled without any discription ,try it if it doesn't work I got one more idea for aizoban.


Reply guys what do u think

**Comment by Adamlord1** [comment]
Anyway guys can u explain in simpler terms what all things are missing in current app.

**Comment by jparkie** [comment]
I'll definitely upload the current progress of Version 2.0. It has all the major components completed; it just requires some user interface over its functions. The user interfaces which are left are MangaActivity, ChapterActivity, and all the DialogFragments.

I don't know if I would come back. Maintaining this application took a lot of work. Originally, I made this application because I didn't want to pay for premium applications. However, as a student who will be starting university this fall, I could not nor was it reasonable for me to try to fund the application to host its own server for advanced features and up-to-date databases. Thus, I really tried to make do and make the least amount of compromises (understanding this was my first application).

Nonetheless, although I got some good feedback, I constantly got hate message for features I stated I couldn't implement or something which was extremely unreasonable for me to do. A good number of those messages were very profane complaining for example like, "Application X has this feature! Why don't you have it?! Stop being a lazy ****, and *******...." I constantly had to reply and apologize that unlike that application, I am not a company who can support the application nor do I have the resources to fund cloud sync, update notification, or general servers while I didn't want all the legal and privacy issues of incorporating the application for monetization because this application is a hobby and I am a student. I understand if they were mad if they were conned for a product, but this application is free without premium features, no advertisements, nor user analytics for any revenue while it still being open-sourced and not encrypted for transparency. Eventually when the application got suspended, I didn't want to try to appeal and risking my account being terminated for this kind of stress and treatment. 

Personally, I do like Android development, and I do still work on various projects and other work which is personally satisfying with non-harassing users, but I wouldn't want to go through the process of releasing a new manga application. Nevertheless, if anyone is willing to take this project, I'll support them; the purpose for Version 2.0 was to make it to modify. I even planned to include testing and the like. Yet, it is incomplete, so I'm sorry. The license for this project is very liberal, so if people ask me I can help them transition it to their own application.

**Comment by Adamlord1** [comment]
Well I am in the same boat will be going to university this fall too.
So did u learn android development on ur own.
That's pretty awesome.
I understand ur problem u know we talked about this before too .the fact abt funding etc .
I understand what ur problem is ,that's why I help u all because I can't  contributute that company class level u talked abt I m in same boat hahaha .I am just a student too doing it for hobby.
But I know what troubles u ,is what troubles me too."how can a simple student go all out on something this big well I have a carrier and study to take care abt" this thought comes in my mind too.

But doing this make me realize one thing that at least I m not betting the bush ,I m doing something new that is different than what others think.making my own path.
Its tough challenging.I m not talking abt aizoban only its a general point.

That's why I would like u to not stop here .
Let's make aizoban not a company app but a joint hobby class by group of lovers of mangas .

U can't do financing ,advetisement etc I understand so can't I .but maybe someone in our family can .
Say brother chogos maybe able to do some things we can't  .

We should contribute what we can.let's not stop today keep moving keep living our hobbies maybe not all over but a small part of our life is worth contributing to something like this.

I would like AIZOBAN to not be a app made by just jparkie but an app run by whole family of lovers of manga which started with jparkie.

U might not be able to do  mangament things but u r very well able to contribute ur mind of a developer right.

NO NEED TO MAKE IT A TOP CLASS PRIORITY LETS CONTINUE THIS APP AS AN HOBBY AND AS MANGA LOVER FAMILY.
IF U CANT GATHER FUNDS ETC FOR THIS MAYBE SOMEONE IN FAMILY CAN.
LETS NOT DROP IT 
PLEASE KEEP GOING LETS NOT LET AIZOBAN END HERE.

OK BROTHER

**Comment by qopisbel** [comment]
@jparkie I'm sorry to hear about your experience with abusive feedback. There is absolutely no excuse for that kind of behavior and it can unfortunately discourage people from developing free and open source software. (Why spend time on something when people give you sh** for it?)

By releasing your software under a permissive license you have made a great contribution to the community and it's disappointing when some members can't appreciate that. There is nothing to apologize for, you have no obligation to continue development or fulfill requests especially from rude people. As the above comments above have shown, your development has attracted much interest and there are individuals willing to contribute in a positive way. This is what I find so exciting about free software projects.

Whether you choose to develop on your own, as a group, or end now, releasing your code as free software was the right choice.

*A sincere thank you from the free software community*

**Comment by Adamlord1** [comment]
That's correct brother

**Comment by butterfist** [comment]
I'm so sorry that this app had been pulled from the play store, i was doing my best to make people know it and they all were amazed by it, so it really is a shame. 

I'm sorry to hear about the bad feedbacks and I hope to see more brilliant work from u!

**Comment by jparkie** [comment]
I added the current progress of Version 2.0.0 as the "incomplete-v2" branch at https://github.com/jparkie/Aizoban/tree/incomplete-v2.

**Comment by KaitoKito** [comment]
jparkie, I want to thank you again for your work, I rally like to read scanlations though Aizoban, not to mention the ease of downloading last chapters (despite an odd way to name folders sorting the files), which keep them safe if scanlations website are taken down someday.

I don't know shit about android development, but I'll be glad to translate Aizoban in french if you want. Have you anticipated future translations and put every words on a lang.xml file or something like this ?

**Comment by BustyLoli-Chan** [comment]
I'm taking this. I don't know where it will go or where it will it end up. I don't even know if I'll follow your visions for 2.0, honestly, I haven't even looked at them. What I do know is that every other manga reader I can find right now is kind of a piece of crap, and this application is absolutely beautiful. It is 3:31 am in the morning and I have class at 10am tomorrow, and I am just so awestruck and inspired by this piece of software. Thank you.

**Comment by Chogos** [comment]
@BustyLoli-Chan I've been working a little on Jacob's V2, but recently I haven't really found time for it and I don't know if I'm going to continue with it or not.. I believe that only the three dialogs and the colour overlay for MangaActivity need to be completed. If you are interested you can find it [here] (https://github.com/Chogos/Aizoban/commit/cb7b1b03247048f964e2c5dd5553f20cc1ebd326).

**Comment by BustyLoli-Chan** [comment]
I will take a look at that, thank you very much!

**Comment by IpodHero188** [comment]
@Chogos Is your version of 2.0 runnable? I have been doing some small tweaks on version 1.0 (was not aware of a 2.0) and the one fix that I'm stuck on in 1.0 is the centering of the page when the screen is set horizontally.

 I know that @jparkie was aware of this in 1.0 and said that he would change how the pages are displayed in 2.0, but I wasn't sure if he coded that in 2.0. I just saw your comment about you continuing 2.0, so I wanted to know if your version is runnable and if the fix about the page centering is fixed. Thanks for your continued effort.

Also if you're reading this @jparkie, I want to thank you for creating this app. It is great and very fun to use.

**Comment by Chogos** [comment]
@IpodHero188 Yes, it is runnable, but I don't remember whether this issue that you mention is resolved or not.

**Comment by shukero** [comment]
Hey everyone, I was just wondering how the development was going on this app. I absolutely love the current version of this app, and I also think this is the best manga app there is for Android. I'm currently using this and I've only found two things that I'm unsure of how to fix:

1) How to add additional sources. As I use www.Mangatown.com for most of my manga

2) How to update those sources. I'm using the mangaeden.com source but since this version of the app came out, a man's series was added to mangaeden that just doesn't show up in the category list.

Can anyone help me out?

Thanks for making such an awesome app. And to be frank, I'm glad this app isn't on Google play. There are some really great manga and manhwa that Google has determined are too "adult" or "ecchi". That is why this app was taken off Google Play.

Anyway thanks again,
Shukero
